### PR TITLE
ACCSEC-19003 Add logs

### PR DESCRIPTION
**Pull request by yafuquen**
## Jira Ticket
- ACCSEC-19003

## Description
- Add logs for error, debug, info and networking

## Commit message
-  Add logs

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

File: TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift [hunk 1]
```diff
@@ -37,53 +37,70 @@ class PushChallengeProcessor {
 
 extension PushChallengeProcessor: PushChallengeProcessorProtocol {
   func getChallenge(withSid sid: String, withFactor factor: PushFactor, success: @escaping ChallengeSuccessBlock, failure: @escaping TwilioVerifyErrorBlock) {
+    Logger.shared.log(withLevel: .info, message: "Getting challenge \(sid) with factor \(factor.sid)")
     challengeProvider.get(withSid: sid, withFactor: factor, success: success, failure: { error in
       failure(TwilioVerifyError.inputError(error: error as NSError))
     })
   }
   
   func updateChallenge(withSid sid: String, withFactor factor: PushFactor, status: ChallengeStatus, success: @escaping EmptySuccessBlock, failure: @escaping TwilioVerifyErrorBlock) {
```
Comment (AuthyApps) [inline TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift:46]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/f45e089425972bef076bacaa943e53d5cf8f3bce/TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift#L46"></span>Function body should span 40 lines or less excluding comments and whitespace: currently spans 51 lines<br />
<code>function_body_length</code> <code>PushChallengeProcessor.swift:46</code></td>
    </tr>
  </tbody>
</table>

File: TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift [hunk 2]
```diff
@@ -43,6 +43,7 @@ extension PushFactory: PushFactoryProtocol {
   func createFactor(withAccessToken accessToken: String, friendlyName: String, pushToken: String, serviceSid: String,
```
Comment (AuthyApps) [inline TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift:43]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/f45e089425972bef076bacaa943e53d5cf8f3bce/TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift#L43"></span>Function body should span 40 lines or less excluding comments and whitespace: currently spans 43 lines<br />
<code>function_body_length</code> <code>PushFactory.swift:43</code></td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by Greshot — APPROVED** [review]

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifyDemo/TwilioVerifyDemo/CreateFactor/Presenter/CreateFactorPresenter.swift [hunk 3]
```diff
@@ -31,14 +31,9 @@ class CreateFactorPresenter {
   private let accessTokensAPI: AccessTokensAPI
   
   init?(withView view: CreateFactorView, accessTokensAPI: AccessTokensAPI = AccessTokensAPIClient()) {
```
Comment (SergioFierro) [inline TwilioVerifyDemo/TwilioVerifyDemo/CreateFactor/Presenter/CreateFactorPresenter.swift:33]: I think you can remove the optional from the init

File: TwilioVerifyDemo/TwilioVerifyDemo/FactorDetail/Presenter/FactorDetailPresenter.swift [hunk 4]
```diff
@@ -39,16 +39,11 @@ class FactorDetailPresenter {
   }
   
   init?(withView view: FactorDetailView?, factor: Factor) {
```
Comment (SergioFierro) [inline TwilioVerifyDemo/TwilioVerifyDemo/FactorDetail/Presenter/FactorDetailPresenter.swift:41]: ditto

File: TwilioVerifyDemo/TwilioVerifyDemo/CreateFactor/Presenter/CreateFactorPresenter.swift (hunk 3 above)
Comment (yafuquen) [reply TwilioVerifyDemo/TwilioVerifyDemo/CreateFactor/Presenter/CreateFactorPresenter.swift:33]: Added a validation for the twilio verify dependency

File: TwilioVerifyDemo/TwilioVerifyDemo/FactorDetail/Presenter/FactorDetailPresenter.swift (hunk 4 above)
Comment (yafuquen) [reply TwilioVerifyDemo/TwilioVerifyDemo/FactorDetail/Presenter/FactorDetailPresenter.swift:41]: Added a validation for the twilio verify dependency

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.18%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerify/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerify/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`97.22%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`92.59%`**
TwilioVerify/TwilioSecurity/Sources/Keychain/Keychain.swift | **`94.37%`**
TwilioVerify/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`91.78%`**
TwilioVerify/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.55%`**
TwilioVerify/TwilioVerify/Sources/API/FactorAPIClient.swift | **`94.20%`**
TwilioVerify/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeListMapper.swift | **`95.83%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeMapper.swift | **`97.67%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerify/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.47%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorMapper.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.80%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
TwilioVerify/TwilioVerify/Sources/Extensions/StringInterpolation+Extensions.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Extensions/URLSession+Extensions.swift | **`92.00%`**
TwilioVerify/TwilioVerify/Sources/Logger/Logger.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Networking/URLRequestBuilder.swift | **`88.73%`**
TwilioVerify/TwilioVerify/Sources/TwilioVerifyError.swift | **`92.42%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SergioFierro — APPROVED** [review]

**Pull request merged**